ts/nni_manager/training_service/reusable/environments/dlcEnvironmentService.ts

```diff
@@ -0,0 +1,146 @@
+export class DlcEnvironmentService extends EnvironmentService {
+
+    private readonly log: Logger = getLogger('dlcEnvironmentService');
+    private experimentId: string;
+    private experimentRootDir: string;
+    private config: FlattenDlcConfig;
+
+    constructor(config: ExperimentConfig, info: ExperimentStartupInfo) {
+        super();
+        this.experimentId = info.experimentId;
+        this.experimentRootDir = info.logDir;
+        this.config = flattenConfig(config, 'dlc');
+        component.Container.bind(StorageService).to(MountedStorageService).scope(Scope.Singleton);
+    }
+
+    public get hasStorageService(): boolean {
+        return false;
+    }
+
+    public initCommandChannel(eventEmitter: EventEmitter): void {
+        this.commandChannel = new FileCommandChannel(eventEmitter);
+    }
+
+    public createEnvironmentInformation(envId: string, envName: string): EnvironmentInformation {
+        return new DlcEnvironmentInformation(envId, envName);
+    }
+
+    public get getName(): string {
+        return 'dlc';
+    }
+
+    public async refreshEnvironmentsStatus(environments: EnvironmentInformation[]): Promise<void> {
+        environments.forEach(async (environment) => {
+            const dlcClient = (environment as DlcEnvironmentInformation).dlcClient;
+            if (!dlcClient) {
+                return Promise.reject('DLC client not initialized!');
+            }
+            const newStatus = await dlcClient.updateStatus(environment.status);
+            switch (newStatus.toUpperCase()) {
+                case 'CREATING':
+                case 'CREATED':
+                case 'WAITING':
+                case 'QUEUED':
+                    environment.setStatus('WAITING');
+                    break;
+                case 'RUNNING':
+                    environment.setStatus('RUNNING');
+                    break;
+                case 'COMPLETED':
+                case 'SUCCEEDED':
+                    environment.setStatus('SUCCEEDED');
+                    break;
+                case 'FAILED':
+                    environment.setStatus('FAILED');
+                    return Promise.reject(`DLC: job ${environment.envId} is failed!`);
+                case 'STOPPED':
+                case 'STOPPING':
+                    environment.setStatus('USER_CANCELED');
+                    break;
+                default:
+                    environment.setStatus('UNKNOWN');
+            }
+        });
+    }
+
+    public async startEnvironment(environment: EnvironmentInformation): Promise<void> {
+        const dlcEnvironment: DlcEnvironmentInformation = environment as DlcEnvironmentInformation;
+        const environmentLocalTempFolder = path.join(this.experimentRootDir, "environment-temp");
+        if (!fs.existsSync(environmentLocalTempFolder)) {
+            await fs.promises.mkdir(environmentLocalTempFolder, {recursive: true});
+        }
+
+        const dlcFolder: string = this.experimentRootDir.replace(
```

Comment:
seems `this.experimentRootDir` is assigned to `info.logDir`, why does it contains `this.config.localStorageMountPoint`? why don't use `dlcFolder = this.config.containerStorageMountPoint` directly?

Reply:
updated to use storageService similar to openPai.